contracts/stellar-interchain-token-service/src/tests/migrate.rs

```diff
@@ -285,305 +363,283 @@ fn migrate_succeeds_with_multiple_token_ids() {
-            &env,
-            token_id_2.clone(),
-            upgrader_client.address,
-            String::from_str(&env, VERSION_V110),
-        ));
-    });
-
-    assert_eq!(
-        env.as_contract(&its_client.address, || {
-            storage::token_manager_wasm_hash(&env)
-        }),
-        token_manager_wasm_hash_v110,
-        "token manager WASM hash should be updated"
+    setup_migrate_storage(
+        &env,
+        token_config_1,
+        &its_client,
+        token_id_1.clone(),
+        current_epoch,
+        flow_in_amount_1,
+        flow_out_amount_1,
     );
-    assert_eq!(
-        env.as_contract(&its_client.address, || {
-            storage::interchain_token_wasm_hash(&env)
-        }),
-        interchain_token_wasm_hash_v110,
-        "interchain token WASM hash should be updated"
-    );
-    assert_eq!(
-        env.as_contract(&its_client.address, || {
-            storage::flow_in(&env, token_id_1.clone(), current_epoch)
-        }),
-        flow_in_amount_1
+
+    setup_migrate_storage(
+        &env,
+        token_config_2,
+        &its_client,
+        token_id_2.clone(),
+        current_epoch,
+        flow_in_amount_2,
+        flow_out_amount_2,
     );
-    assert_eq!(
-        env.as_contract(&its_client.address, || {
-            storage::flow_out(&env, token_id_1, current_epoch)
-        }),
-        flow_out_amount_1
+
+    let its_upgrade_auth = mock_auth!(owner, its_client.upgrade(&its_wasm_hash));
+    let its_migrate_auth = mock_auth!(owner, its_client.migrate(migration_data.clone()));
+    let upgrader_upgrade_auth = mock_auth!(
+        owner,
+        upgrader_client.upgrade(
+            &its_client.address,
+            &String::from_str(&env, NEW_VERSION),
+            &its_wasm_hash,
+            &vec![&env, migration_data.clone()],
+        ),
+        &[
+            its_upgrade_auth.invoke.clone(),
+            its_migrate_auth.invoke.clone()
+        ]
     );
-    assert_eq!(
-        env.as_contract(&its_client.address, || {
-            storage::flow_in(&env, token_id_2.clone(), current_epoch)
-        }),
-        flow_in_amount_2
+
+    upgrader_client
+        .mock_auths(&[upgrader_upgrade_auth])
+        .upgrade(
+            &its_client.address,
+            &String::from_str(&env, NEW_VERSION),
+            &its_wasm_hash,
+            &vec![&env, migration_data.into_val(&env)],
+        );
+
+    its_client
+        // .mock_auths(&[its_migrate_token_auth]) // FIXME: Use
+        .mock_all_auths_allowing_non_root_auth()
+        .migrate_token(
+            &token_id_1,
+            &upgrader_client.address,
+            &String::from_str(&env, NEW_VERSION),
+        );
+
+    its_client
+        // .mock_auths(&[its_migrate_token_auth]) // FIXME: Use
+        .mock_all_auths_allowing_non_root_auth()
+        .migrate_token(
+            &token_id_2,
+            &upgrader_client.address,
+            &String::from_str(&env, NEW_VERSION),
+        );
```

Comment:
now that tokens aren't migrated atomically, this test for migrating multiple tokens doesn't really make sense. There should be separate tests for the two token manager types, but otherwise they're independent